### PR TITLE
D-Agent Changes corresponding to PoseidonOS CLI

### PR DESCRIPTION
**Pull request by aswinssir**
Added the API endpoints for Listing Transport and Deleting Listener Integrated latest Kouros from PoseidonOS

### CODE REVIEW

File: src/dagent/src/routers/m9k/api/caller/array.go [hunk 1]
```diff
@@ -142,7 +142,9 @@ func CallListArray(xrId string, param interface{}, posMngr pos.POSManager) (mode
 		log.Errorf(commandFailureMsg, GetFuncName(1), err1)
 		return model.Response{}, ErrConn
 	}
-	resByte, err2 := protojson.Marshal(result)
+	resByte, err2 := protojson.MarshalOptions{
+		EmitUnpopulated: true,
+	}.Marshal(result)
 	return HandleResponse(resByte, err2)
```
Comment (vishal-shakya-ssir) [inline src/dagent/src/routers/m9k/api/caller/array.go:148]: what is the purpose of below here ?
MarshalOptions{
		EmitUnpopulated: true,
	}.Marshal(result)

**Review by vishal-shakya-ssir — APPROVED** [review]
LGTM

File: src/dagent/src/routers/m9k/api/caller/array.go (hunk 1 above)
Comment (aswinssir) [reply src/dagent/src/routers/m9k/api/caller/array.go:148]: If this is not added, then 0 values, empty strings and null values will not be present in the JSON output

**Review by manoj22-k — APPROVED** [review]
LGTM

**Pull request merged**